text/0087-graphql-errors.md

```diff
@@ -0,0 +1,262 @@
+
+However, the `Response` interface does not contain the `data` field which is the response body.
+
+# Background
+
+This is necessary to implement the [GraphQL Client Errors](https://www.notion.so/sentry/GraphQL-proposal-d6e5846f30434770903cf3af20bc2568) with syntax highlight.
+
+The `Request` and `Response` bodies could contain PII.
+
+Request body example (query language only):
+
+```graphql
+query { 
+  viewer { 
+    login
+  }
+}
+```
+
+Response body example:
+
+```json
+{
+  "data": {
+    "viewer": {
+      "login": "foo"
+    }
+  }
+}
+```
+
+A Request example with an error (query language only):
+
+```graphql
+query { 
+  viewer { 
+    # note the removed `n` at the end
+    logi
+  }
+}
+```
+
+Response body example:
+
+```json
+{
+  "errors": [
+    {
+      "path": [
+        "query",
+        "viewer",
+        "logi"
+      ],
+      "extensions": {
+        "code": "undefinedField",
+        "typeName": "User",
+        "fieldName": "logi"
+      },
+      "locations": [
+        {
+          "line": 4,
+          "column": 5
+        }
+      ],
+      "message": "Field 'logi' doesn't exist on type 'User'"
+    }
+  ]
+}
+```
+
+Using the `locations` and the request body (query language only), we can highlight the error in the request body.
+
+```graphql
+query { 
+  viewer { 
+    # note the removed `n` at the end
+    -> logi
+  }
+}
+```
+
+Request body example, full body (not only the query language):
+
+```json
+{
+  "query": "{\n  viewer {\n    login\n  }\n}",
+  "variables": {}
+}
+```
+
+Request body example, full body (not only the query language):
+
+```json
+{
+  "query": "{\n  viewer {\n    login\n  }\n}",
+  "variables": {}
+}
+```
+
+The Request body can also contain `variables`.
+
+```json
+{
+  "query": "{\n  viewer {\n    login\n  }\n}",
+  "variables": {
+    "login": "foo"
+  }
+}
+```
+
+Because of that, the `Request` and `Response` body should be sent to Sentry.
+
+# Supporting Data
+
+See [JIRA issues](https://www.notion.so/sentry/GraphQL-proposal-d6e5846f30434770903cf3af20bc2568?pvs=4#fd0b946340d947eb9bbaa320846ccd6a), not disclosing here because they could contain PII.
+
+Related issues and discussions:
+
+[GraphQL Support](https://github.com/getsentry/sentry/discussions/38913)
+
+[Support for GraphQL errors](https://github.com/getsentry/sentry/issues/33723)
+
+[Send request body for http client integrations and similar](https://github.com/getsentry/team-mobile/issues/41)
+
+# Proposal (Option 1)
+
+The proposal is adding a `data` field in the [Response](https://develop.sentry.dev/sdk/event-payloads/types/#responsecontext) interface.
+
+By doing this, we can keep the `Request` interface as it is, we can copy the `data` scrubbing rules from the `Request` interface.
+
+```json
+{
+  "contexts": {
+    "response": {
+      "type": "response",
+      "status_code": 500,
+      "body_size": 1000,
+      "data": "foo",
```

Comment:
`l`: I guess most of the time it will be something similar to the following. Would make the RFC easier to understand, IMO.
```suggestion
    "data": {
      "foo": "bar"
    },
```